packages/editor-ui/src/components/ResourceMapper/ResourceMapper.vue

```diff
@@ -244,25 +244,45 @@ async function loadFieldsToMap(): Promise<void> {
 		return;
 	}
 
-	const methodName = props.parameter.typeOptions?.resourceMapper?.resourceMapperMethod;
-	if (typeof methodName !== 'string') {
-		return;
+	const resourceMapperMethod = props.parameter.typeOptions?.resourceMapper?.resourceMapperMethod;
+	const localResourceMapperMethod =
+		props.parameter.typeOptions?.resourceMapper?.localResourceMapperMethod;
+
+	let fetchedFields = null;
+
+	if (typeof resourceMapperMethod === 'string') {
```

Comment:
nit: Since this is a bit of a weird pattern could we extract this into a `fetchFields` method that resolves unique properties and callback depending on the input and throws if both are defined? I know we have the type-level protection but I'd rather we crash early if someone avoids it, rather than silently going with the non-local version. E.g.:

```typescript
function resolveLocal(localThing: LocalThing) {
  return { callback: nodeTypesStore.getResourceMapperFields, props: { methodName: localThing } };
}

function resolveExternal(externalThing: ExternalThing) {
    return { callback: nodeTypesStore.getResourceMapperFields, props: { methodName: externalThing, credentials: props.node.credentials } };
}

function fetchFields(method: ResourceMapperMethod) {
    const { a = undefined , b = undefined = method;
    if (a && b) throw ...;
    const { callback, props } = a ? resolveLocal(a) : resolveExternal(b);

    const params : RMFR = {
        {}, // <the other stuff that's the same>
        ...props
    } ;

    return await callback(props);
}
```

I appreciate that's not a small refactor, so feel free to skip or do as a follow up if you do like it but want to land this first